src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -213,27 +221,39 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                     throw;
                 }
 
-                Debug.Assert(!Monitor.IsEntered(SyncObject));
-                lock (SyncObject)
+                // New scope here to avoid variable name conflict on "sendReset"
```

Comment:
Did you consider instead just doing:
```
sendReset = false;
```
instead of:
```
bool sendReset = false;
```
?

Reply:
The scope conflict here is with sendReset in the catch block.

I suppose I could hoist sendReset out of the try/catch entirely, but I'm not sure it matters much really.